In [1]:
import pandas as pd
import numpy as np
from catboost import CatBoostClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import cohen_kappa_score
from xgboost import XGBClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from imblearn.ensemble import BalancedBaggingClassifier
from sklearn.naive_bayes import GaussianNB

In [2]:
df1 = pd.read_csv("final_train_al.csv",index_col=0)
df2 = pd.read_csv("final_train_pl.csv",index_col=0)
df3 = pd.read_csv("final_train_ll.csv",index_col=0)

In [3]:
X1= df1.drop(['class'], axis=1)
Y1 = df1['class']
X2= df2.drop(['class'], axis=1)
Y2 = df2['class']
X3= df3.drop(['class'], axis=1)
Y3 = df3['class']

In [4]:
X1_train, X1_test, Y1_train, Y1_test = train_test_split(X1, Y1, test_size = 0.2)
X2_train, X2_test, Y2_train, Y2_test = train_test_split(X2, Y2, test_size = 0.2)
X3_train, X3_test, Y3_train, Y3_test = train_test_split(X3, Y3, test_size = 0.2)

In [20]:
# define models
# Logistic Regression classifier(lambda = 0.1, 1, 10,100)
lr = LogisticRegression(penalty = 'l2', random_state = 0)

# Random Forest Classifier
rf = RandomForestClassifier(n_estimators = 90, oob_score = True, criterion = 'entropy')

# eXtreme Gradient Boosting Classifier (XGBClassifier)
xgb = XGBClassifier(max_depth=20, n_estimators=200,objective='multi:softmax',num_class=3)

# Balanced Bagging Classifier
bb = BalancedBaggingClassifier(base_estimator = RandomForestClassifier(criterion='gini'), n_estimators = 30, bootstrap = True)

# CatBoost Classifier
cc = CatBoostClassifier(iterations=1000, depth=8, learning_rate=0.2)

# Naive Bayesian
gn = GaussianNB()

In [21]:
score = {'lr':[],'rf':[],'xgb':[],'bb':[],'cc':[],'gn':[]}
for data in [df1,df2,df3]:
    X1= data.drop(['class'], axis=1)
    Y1 = data['class']
    X1_train, X1_test, Y1_train, Y1_test = train_test_split(X1, Y1, test_size = 0.2)
    lr.fit(X1_train, Y1_train)
    y_predict = lr.predict(X1_test)
    score['lr'].append(cohen_kappa_score(Y1_test,y_predict))
    rf.fit(X1_train, Y1_train)
    y_predict = rf.predict(X1_test)
    score['rf'].append(cohen_kappa_score(Y1_test,y_predict))
    xgb.fit(X1_train, Y1_train)
    y_predict = xgb.predict(X1_test)
    score['xgb'].append(cohen_kappa_score(Y1_test,y_predict))
    bb.fit(X1_train, Y1_train)
    y_predict = bb.predict(X1_test)
    score['bb'].append(cohen_kappa_score(Y1_test,y_predict))
    cc.fit(X1_train, Y1_train)
    y_predict = cc.predict(X1_test)
    score['cc'].append(cohen_kappa_score(Y1_test,y_predict))
    gn.fit(X1_train, Y1_train)
    y_predict = gn.predict(X1_test)
    score['gn'].append(cohen_kappa_score(Y1_test,y_predict))

/Users/zionzhou/opt/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)


0:	learn: 0.9613061	total: 13ms	remaining: 13s
1:	learn: 0.8541353	total: 25ms	remaining: 12.5s
2:	learn: 0.7844204	total: 36.2ms	remaining: 12s
3:	learn: 0.7274817	total: 47.7ms	remaining: 11.9s
4:	learn: 0.6856375	total: 58.8ms	remaining: 11.7s
5:	learn: 0.6498382	total: 70.1ms	remaining: 11.6s
6:	learn: 0.6235923	total: 81.4ms	remaining: 11.5s
7:	learn: 0.5986059	total: 92.5ms	remaining: 11.5s
8:	learn: 0.5838875	total: 105ms	remaining: 11.5s
9:	learn: 0.5699785	total: 117ms	remaining: 11.6s
10:	learn: 0.5577693	total: 129ms	remaining: 11.6s
11:	learn: 0.5465266	total: 141ms	remaining: 11.6s
12:	learn: 0.5341841	total: 152ms	remaining: 11.5s
13:	learn: 0.5259158	total: 165ms	remaining: 11.6s
14:	learn: 0.5203894	total: 178ms	remaining: 11.7s
15:	learn: 0.5110433	total: 190ms	remaining: 11.7s
16:	learn: 0.5053546	total: 203ms	remaining: 11.7s
17:	learn: 0.5015768	total: 217ms	remaining: 11.9s
18:	learn: 0.4967373	total: 235ms	remaining: 12.1s
19:	learn: 0.4934127	total: 248ms	remaini

172:	learn: 0.2466838	total: 2.29s	remaining: 10.9s
173:	learn: 0.2454996	total: 2.3s	remaining: 10.9s
174:	learn: 0.2444483	total: 2.32s	remaining: 10.9s
175:	learn: 0.2431164	total: 2.33s	remaining: 10.9s
176:	learn: 0.2427402	total: 2.34s	remaining: 10.9s
177:	learn: 0.2415587	total: 2.35s	remaining: 10.9s
178:	learn: 0.2411299	total: 2.37s	remaining: 10.8s
179:	learn: 0.2403885	total: 2.38s	remaining: 10.8s
180:	learn: 0.2394084	total: 2.39s	remaining: 10.8s
181:	learn: 0.2385519	total: 2.41s	remaining: 10.8s
182:	learn: 0.2373193	total: 2.43s	remaining: 10.8s
183:	learn: 0.2369079	total: 2.44s	remaining: 10.8s
184:	learn: 0.2363844	total: 2.46s	remaining: 10.8s
185:	learn: 0.2356931	total: 2.48s	remaining: 10.8s
186:	learn: 0.2345458	total: 2.49s	remaining: 10.8s
187:	learn: 0.2341300	total: 2.5s	remaining: 10.8s
188:	learn: 0.2331495	total: 2.52s	remaining: 10.8s
189:	learn: 0.2323869	total: 2.53s	remaining: 10.8s
190:	learn: 0.2320208	total: 2.54s	remaining: 10.8s
191:	learn: 0.

343:	learn: 0.1493610	total: 4.56s	remaining: 8.7s
344:	learn: 0.1490982	total: 4.57s	remaining: 8.68s
345:	learn: 0.1486508	total: 4.58s	remaining: 8.66s
346:	learn: 0.1480841	total: 4.59s	remaining: 8.65s
347:	learn: 0.1478296	total: 4.61s	remaining: 8.63s
348:	learn: 0.1474138	total: 4.62s	remaining: 8.62s
349:	learn: 0.1469297	total: 4.63s	remaining: 8.6s
350:	learn: 0.1464840	total: 4.64s	remaining: 8.58s
351:	learn: 0.1459847	total: 4.65s	remaining: 8.57s
352:	learn: 0.1455590	total: 4.67s	remaining: 8.55s
353:	learn: 0.1454668	total: 4.68s	remaining: 8.54s
354:	learn: 0.1451419	total: 4.69s	remaining: 8.52s
355:	learn: 0.1450172	total: 4.7s	remaining: 8.5s
356:	learn: 0.1444627	total: 4.71s	remaining: 8.49s
357:	learn: 0.1438350	total: 4.72s	remaining: 8.47s
358:	learn: 0.1434745	total: 4.74s	remaining: 8.45s
359:	learn: 0.1430400	total: 4.75s	remaining: 8.44s
360:	learn: 0.1427081	total: 4.76s	remaining: 8.42s
361:	learn: 0.1423668	total: 4.77s	remaining: 8.41s
362:	learn: 0.14

503:	learn: 0.1019982	total: 6.61s	remaining: 6.5s
504:	learn: 0.1017596	total: 6.62s	remaining: 6.49s
505:	learn: 0.1015905	total: 6.63s	remaining: 6.47s
506:	learn: 0.1013381	total: 6.64s	remaining: 6.46s
507:	learn: 0.1011831	total: 6.66s	remaining: 6.45s
508:	learn: 0.1009851	total: 6.67s	remaining: 6.43s
509:	learn: 0.1008380	total: 6.68s	remaining: 6.42s
510:	learn: 0.1006053	total: 6.69s	remaining: 6.41s
511:	learn: 0.1004418	total: 6.71s	remaining: 6.39s
512:	learn: 0.1002667	total: 6.72s	remaining: 6.38s
513:	learn: 0.1001832	total: 6.73s	remaining: 6.36s
514:	learn: 0.0999495	total: 6.75s	remaining: 6.35s
515:	learn: 0.0997130	total: 6.76s	remaining: 6.34s
516:	learn: 0.0995005	total: 6.77s	remaining: 6.32s
517:	learn: 0.0993577	total: 6.78s	remaining: 6.31s
518:	learn: 0.0991962	total: 6.79s	remaining: 6.3s
519:	learn: 0.0991208	total: 6.81s	remaining: 6.29s
520:	learn: 0.0989494	total: 6.82s	remaining: 6.27s
521:	learn: 0.0987401	total: 6.83s	remaining: 6.26s
522:	learn: 0.

663:	learn: 0.0750492	total: 8.68s	remaining: 4.39s
664:	learn: 0.0748879	total: 8.69s	remaining: 4.38s
665:	learn: 0.0748186	total: 8.71s	remaining: 4.37s
666:	learn: 0.0747111	total: 8.72s	remaining: 4.36s
667:	learn: 0.0746147	total: 8.74s	remaining: 4.34s
668:	learn: 0.0744952	total: 8.76s	remaining: 4.33s
669:	learn: 0.0743926	total: 8.77s	remaining: 4.32s
670:	learn: 0.0742368	total: 8.78s	remaining: 4.31s
671:	learn: 0.0741569	total: 8.8s	remaining: 4.29s
672:	learn: 0.0740349	total: 8.81s	remaining: 4.28s
673:	learn: 0.0738734	total: 8.82s	remaining: 4.27s
674:	learn: 0.0736748	total: 8.84s	remaining: 4.25s
675:	learn: 0.0736067	total: 8.85s	remaining: 4.24s
676:	learn: 0.0735186	total: 8.86s	remaining: 4.23s
677:	learn: 0.0734574	total: 8.87s	remaining: 4.21s
678:	learn: 0.0732872	total: 8.89s	remaining: 4.2s
679:	learn: 0.0731131	total: 8.91s	remaining: 4.19s
680:	learn: 0.0729596	total: 8.92s	remaining: 4.18s
681:	learn: 0.0727299	total: 8.94s	remaining: 4.17s
682:	learn: 0.

837:	learn: 0.0567923	total: 10.9s	remaining: 2.11s
838:	learn: 0.0567465	total: 10.9s	remaining: 2.1s
839:	learn: 0.0566611	total: 10.9s	remaining: 2.08s
840:	learn: 0.0566277	total: 11s	remaining: 2.07s
841:	learn: 0.0565224	total: 11s	remaining: 2.06s
842:	learn: 0.0564262	total: 11s	remaining: 2.04s
843:	learn: 0.0563143	total: 11s	remaining: 2.03s
844:	learn: 0.0562221	total: 11s	remaining: 2.02s
845:	learn: 0.0561423	total: 11s	remaining: 2.01s
846:	learn: 0.0560086	total: 11s	remaining: 1.99s
847:	learn: 0.0559536	total: 11.1s	remaining: 1.98s
848:	learn: 0.0558066	total: 11.1s	remaining: 1.97s
849:	learn: 0.0557320	total: 11.1s	remaining: 1.95s
850:	learn: 0.0556898	total: 11.1s	remaining: 1.94s
851:	learn: 0.0555914	total: 11.1s	remaining: 1.93s
852:	learn: 0.0555388	total: 11.1s	remaining: 1.91s
853:	learn: 0.0554740	total: 11.1s	remaining: 1.9s
854:	learn: 0.0554089	total: 11.1s	remaining: 1.89s
855:	learn: 0.0553480	total: 11.1s	remaining: 1.88s
856:	learn: 0.0552700	total:

/Users/zionzhou/opt/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)


0:	learn: 0.9635570	total: 14.9ms	remaining: 14.9s
1:	learn: 0.8652623	total: 28.2ms	remaining: 14.1s
2:	learn: 0.8011322	total: 42.5ms	remaining: 14.1s
3:	learn: 0.7455932	total: 56.4ms	remaining: 14.1s
4:	learn: 0.7057382	total: 69ms	remaining: 13.7s
5:	learn: 0.6758435	total: 81.3ms	remaining: 13.5s
6:	learn: 0.6524007	total: 93.9ms	remaining: 13.3s
7:	learn: 0.6259118	total: 106ms	remaining: 13.1s
8:	learn: 0.6084205	total: 118ms	remaining: 13s
9:	learn: 0.5901296	total: 130ms	remaining: 12.9s
10:	learn: 0.5768125	total: 142ms	remaining: 12.8s
11:	learn: 0.5662742	total: 155ms	remaining: 12.8s
12:	learn: 0.5575871	total: 168ms	remaining: 12.8s
13:	learn: 0.5507314	total: 180ms	remaining: 12.7s
14:	learn: 0.5415449	total: 193ms	remaining: 12.6s
15:	learn: 0.5338303	total: 205ms	remaining: 12.6s
16:	learn: 0.5242624	total: 219ms	remaining: 12.7s
17:	learn: 0.5182818	total: 233ms	remaining: 12.7s
18:	learn: 0.5136736	total: 245ms	remaining: 12.7s
19:	learn: 0.5098127	total: 258ms	rema

161:	learn: 0.2743110	total: 2.08s	remaining: 10.8s
162:	learn: 0.2731374	total: 2.09s	remaining: 10.7s
163:	learn: 0.2722456	total: 2.1s	remaining: 10.7s
164:	learn: 0.2715321	total: 2.12s	remaining: 10.7s
165:	learn: 0.2706047	total: 2.13s	remaining: 10.7s
166:	learn: 0.2693236	total: 2.14s	remaining: 10.7s
167:	learn: 0.2684938	total: 2.15s	remaining: 10.7s
168:	learn: 0.2678429	total: 2.17s	remaining: 10.6s
169:	learn: 0.2669653	total: 2.18s	remaining: 10.6s
170:	learn: 0.2659569	total: 2.19s	remaining: 10.6s
171:	learn: 0.2651091	total: 2.2s	remaining: 10.6s
172:	learn: 0.2639662	total: 2.21s	remaining: 10.6s
173:	learn: 0.2631720	total: 2.23s	remaining: 10.6s
174:	learn: 0.2620766	total: 2.24s	remaining: 10.6s
175:	learn: 0.2606808	total: 2.25s	remaining: 10.5s
176:	learn: 0.2599351	total: 2.26s	remaining: 10.5s
177:	learn: 0.2591844	total: 2.28s	remaining: 10.5s
178:	learn: 0.2583848	total: 2.29s	remaining: 10.5s
179:	learn: 0.2573356	total: 2.3s	remaining: 10.5s
180:	learn: 0.2

324:	learn: 0.1732191	total: 4.12s	remaining: 8.56s
325:	learn: 0.1725252	total: 4.13s	remaining: 8.55s
326:	learn: 0.1722158	total: 4.15s	remaining: 8.54s
327:	learn: 0.1717930	total: 4.16s	remaining: 8.53s
328:	learn: 0.1712636	total: 4.18s	remaining: 8.52s
329:	learn: 0.1707333	total: 4.19s	remaining: 8.5s
330:	learn: 0.1704355	total: 4.2s	remaining: 8.49s
331:	learn: 0.1697990	total: 4.21s	remaining: 8.47s
332:	learn: 0.1694898	total: 4.22s	remaining: 8.46s
333:	learn: 0.1689722	total: 4.24s	remaining: 8.45s
334:	learn: 0.1686188	total: 4.25s	remaining: 8.44s
335:	learn: 0.1680795	total: 4.26s	remaining: 8.42s
336:	learn: 0.1677314	total: 4.27s	remaining: 8.41s
337:	learn: 0.1672031	total: 4.29s	remaining: 8.39s
338:	learn: 0.1667733	total: 4.3s	remaining: 8.38s
339:	learn: 0.1663195	total: 4.31s	remaining: 8.37s
340:	learn: 0.1661401	total: 4.32s	remaining: 8.36s
341:	learn: 0.1658310	total: 4.34s	remaining: 8.34s
342:	learn: 0.1653387	total: 4.35s	remaining: 8.33s
343:	learn: 0.1

498:	learn: 0.1138793	total: 6.36s	remaining: 6.39s
499:	learn: 0.1137645	total: 6.37s	remaining: 6.37s
500:	learn: 0.1136849	total: 6.39s	remaining: 6.36s
501:	learn: 0.1134903	total: 6.4s	remaining: 6.35s
502:	learn: 0.1132224	total: 6.41s	remaining: 6.34s
503:	learn: 0.1129926	total: 6.43s	remaining: 6.32s
504:	learn: 0.1128232	total: 6.44s	remaining: 6.31s
505:	learn: 0.1124926	total: 6.45s	remaining: 6.3s
506:	learn: 0.1122695	total: 6.47s	remaining: 6.29s
507:	learn: 0.1120938	total: 6.48s	remaining: 6.28s
508:	learn: 0.1119872	total: 6.49s	remaining: 6.26s
509:	learn: 0.1118196	total: 6.51s	remaining: 6.26s
510:	learn: 0.1115448	total: 6.53s	remaining: 6.25s
511:	learn: 0.1112871	total: 6.54s	remaining: 6.24s
512:	learn: 0.1112183	total: 6.55s	remaining: 6.22s
513:	learn: 0.1111190	total: 6.57s	remaining: 6.21s
514:	learn: 0.1109060	total: 6.58s	remaining: 6.2s
515:	learn: 0.1106545	total: 6.6s	remaining: 6.19s
516:	learn: 0.1104765	total: 6.61s	remaining: 6.18s
517:	learn: 0.11

673:	learn: 0.0815453	total: 8.59s	remaining: 4.16s
674:	learn: 0.0813745	total: 8.61s	remaining: 4.14s
675:	learn: 0.0812846	total: 8.62s	remaining: 4.13s
676:	learn: 0.0811604	total: 8.63s	remaining: 4.12s
677:	learn: 0.0810945	total: 8.64s	remaining: 4.11s
678:	learn: 0.0809900	total: 8.66s	remaining: 4.09s
679:	learn: 0.0808696	total: 8.67s	remaining: 4.08s
680:	learn: 0.0807303	total: 8.68s	remaining: 4.07s
681:	learn: 0.0805528	total: 8.7s	remaining: 4.05s
682:	learn: 0.0804223	total: 8.71s	remaining: 4.04s
683:	learn: 0.0802287	total: 8.72s	remaining: 4.03s
684:	learn: 0.0800586	total: 8.73s	remaining: 4.01s
685:	learn: 0.0799599	total: 8.74s	remaining: 4s
686:	learn: 0.0798517	total: 8.76s	remaining: 3.99s
687:	learn: 0.0797311	total: 8.77s	remaining: 3.98s
688:	learn: 0.0795724	total: 8.78s	remaining: 3.96s
689:	learn: 0.0794278	total: 8.79s	remaining: 3.95s
690:	learn: 0.0792457	total: 8.81s	remaining: 3.94s
691:	learn: 0.0791470	total: 8.82s	remaining: 3.92s
692:	learn: 0.07

837:	learn: 0.0627295	total: 10.6s	remaining: 2.06s
838:	learn: 0.0626597	total: 10.7s	remaining: 2.04s
839:	learn: 0.0625875	total: 10.7s	remaining: 2.03s
840:	learn: 0.0625074	total: 10.7s	remaining: 2.02s
841:	learn: 0.0624567	total: 10.7s	remaining: 2.01s
842:	learn: 0.0623856	total: 10.7s	remaining: 1.99s
843:	learn: 0.0623079	total: 10.7s	remaining: 1.98s
844:	learn: 0.0622729	total: 10.7s	remaining: 1.97s
845:	learn: 0.0621539	total: 10.7s	remaining: 1.96s
846:	learn: 0.0620766	total: 10.8s	remaining: 1.94s
847:	learn: 0.0619704	total: 10.8s	remaining: 1.93s
848:	learn: 0.0619093	total: 10.8s	remaining: 1.92s
849:	learn: 0.0617883	total: 10.8s	remaining: 1.9s
850:	learn: 0.0616480	total: 10.8s	remaining: 1.89s
851:	learn: 0.0615840	total: 10.8s	remaining: 1.88s
852:	learn: 0.0614775	total: 10.8s	remaining: 1.87s
853:	learn: 0.0613411	total: 10.8s	remaining: 1.85s
854:	learn: 0.0612494	total: 10.9s	remaining: 1.84s
855:	learn: 0.0611612	total: 10.9s	remaining: 1.83s
856:	learn: 0

999:	learn: 0.0497390	total: 12.7s	remaining: 0us


/Users/zionzhou/opt/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)


0:	learn: 0.9545873	total: 14.2ms	remaining: 14.2s
1:	learn: 0.8562818	total: 27.1ms	remaining: 13.5s
2:	learn: 0.7894714	total: 39.1ms	remaining: 13s
3:	learn: 0.7428343	total: 51.5ms	remaining: 12.8s
4:	learn: 0.7004131	total: 63.4ms	remaining: 12.6s
5:	learn: 0.6670419	total: 76.1ms	remaining: 12.6s
6:	learn: 0.6425193	total: 88.5ms	remaining: 12.6s
7:	learn: 0.6196940	total: 101ms	remaining: 12.6s
8:	learn: 0.6012705	total: 116ms	remaining: 12.7s
9:	learn: 0.5870923	total: 130ms	remaining: 12.8s
10:	learn: 0.5718680	total: 142ms	remaining: 12.8s
11:	learn: 0.5617156	total: 154ms	remaining: 12.7s
12:	learn: 0.5520382	total: 166ms	remaining: 12.6s
13:	learn: 0.5445627	total: 178ms	remaining: 12.6s
14:	learn: 0.5384898	total: 191ms	remaining: 12.5s
15:	learn: 0.5322592	total: 203ms	remaining: 12.5s
16:	learn: 0.5234894	total: 217ms	remaining: 12.5s
17:	learn: 0.5175936	total: 230ms	remaining: 12.6s
18:	learn: 0.5122329	total: 243ms	remaining: 12.5s
19:	learn: 0.5061226	total: 255ms	re

175:	learn: 0.2661924	total: 2.27s	remaining: 10.6s
176:	learn: 0.2653662	total: 2.28s	remaining: 10.6s
177:	learn: 0.2649350	total: 2.3s	remaining: 10.6s
178:	learn: 0.2640808	total: 2.31s	remaining: 10.6s
179:	learn: 0.2633711	total: 2.32s	remaining: 10.6s
180:	learn: 0.2625709	total: 2.33s	remaining: 10.6s
181:	learn: 0.2619877	total: 2.35s	remaining: 10.5s
182:	learn: 0.2613337	total: 2.36s	remaining: 10.5s
183:	learn: 0.2608676	total: 2.37s	remaining: 10.5s
184:	learn: 0.2603465	total: 2.38s	remaining: 10.5s
185:	learn: 0.2597157	total: 2.4s	remaining: 10.5s
186:	learn: 0.2585340	total: 2.41s	remaining: 10.5s
187:	learn: 0.2577176	total: 2.42s	remaining: 10.5s
188:	learn: 0.2567704	total: 2.43s	remaining: 10.4s
189:	learn: 0.2560684	total: 2.45s	remaining: 10.4s
190:	learn: 0.2553705	total: 2.46s	remaining: 10.4s
191:	learn: 0.2549121	total: 2.48s	remaining: 10.4s
192:	learn: 0.2541868	total: 2.49s	remaining: 10.4s
193:	learn: 0.2533612	total: 2.5s	remaining: 10.4s
194:	learn: 0.2

348:	learn: 0.1623041	total: 4.54s	remaining: 8.46s
349:	learn: 0.1618544	total: 4.55s	remaining: 8.45s
350:	learn: 0.1614064	total: 4.56s	remaining: 8.44s
351:	learn: 0.1607754	total: 4.58s	remaining: 8.42s
352:	learn: 0.1604241	total: 4.59s	remaining: 8.41s
353:	learn: 0.1602905	total: 4.6s	remaining: 8.4s
354:	learn: 0.1597164	total: 4.62s	remaining: 8.39s
355:	learn: 0.1592755	total: 4.63s	remaining: 8.37s
356:	learn: 0.1588024	total: 4.64s	remaining: 8.36s
357:	learn: 0.1584769	total: 4.66s	remaining: 8.35s
358:	learn: 0.1582623	total: 4.67s	remaining: 8.34s
359:	learn: 0.1577146	total: 4.68s	remaining: 8.32s
360:	learn: 0.1569481	total: 4.69s	remaining: 8.31s
361:	learn: 0.1565443	total: 4.71s	remaining: 8.3s
362:	learn: 0.1562281	total: 4.72s	remaining: 8.29s
363:	learn: 0.1556146	total: 4.74s	remaining: 8.27s
364:	learn: 0.1552996	total: 4.75s	remaining: 8.26s
365:	learn: 0.1548177	total: 4.76s	remaining: 8.25s
366:	learn: 0.1542729	total: 4.78s	remaining: 8.23s
367:	learn: 0.1

515:	learn: 0.1085628	total: 6.77s	remaining: 6.35s
516:	learn: 0.1082484	total: 6.78s	remaining: 6.34s
517:	learn: 0.1079544	total: 6.8s	remaining: 6.33s
518:	learn: 0.1077568	total: 6.81s	remaining: 6.31s
519:	learn: 0.1075610	total: 6.82s	remaining: 6.3s
520:	learn: 0.1073467	total: 6.84s	remaining: 6.29s
521:	learn: 0.1071326	total: 6.85s	remaining: 6.27s
522:	learn: 0.1069411	total: 6.86s	remaining: 6.26s
523:	learn: 0.1067425	total: 6.88s	remaining: 6.25s
524:	learn: 0.1064696	total: 6.89s	remaining: 6.24s
525:	learn: 0.1062795	total: 6.91s	remaining: 6.22s
526:	learn: 0.1061236	total: 6.92s	remaining: 6.21s
527:	learn: 0.1059017	total: 6.93s	remaining: 6.2s
528:	learn: 0.1056749	total: 6.94s	remaining: 6.18s
529:	learn: 0.1054995	total: 6.96s	remaining: 6.17s
530:	learn: 0.1052951	total: 6.97s	remaining: 6.16s
531:	learn: 0.1051714	total: 6.99s	remaining: 6.14s
532:	learn: 0.1049683	total: 7s	remaining: 6.13s
533:	learn: 0.1047249	total: 7.01s	remaining: 6.12s
534:	learn: 0.1045

686:	learn: 0.0781207	total: 9.03s	remaining: 4.11s
687:	learn: 0.0779575	total: 9.04s	remaining: 4.1s
688:	learn: 0.0778787	total: 9.06s	remaining: 4.09s
689:	learn: 0.0777831	total: 9.07s	remaining: 4.07s
690:	learn: 0.0777060	total: 9.08s	remaining: 4.06s
691:	learn: 0.0776182	total: 9.09s	remaining: 4.05s
692:	learn: 0.0774140	total: 9.11s	remaining: 4.03s
693:	learn: 0.0772947	total: 9.12s	remaining: 4.02s
694:	learn: 0.0772280	total: 9.13s	remaining: 4.01s
695:	learn: 0.0771131	total: 9.14s	remaining: 3.99s
696:	learn: 0.0770494	total: 9.15s	remaining: 3.98s
697:	learn: 0.0770060	total: 9.17s	remaining: 3.97s
698:	learn: 0.0768968	total: 9.18s	remaining: 3.95s
699:	learn: 0.0767672	total: 9.19s	remaining: 3.94s
700:	learn: 0.0766306	total: 9.21s	remaining: 3.93s
701:	learn: 0.0765007	total: 9.22s	remaining: 3.91s
702:	learn: 0.0764033	total: 9.23s	remaining: 3.9s
703:	learn: 0.0762676	total: 9.24s	remaining: 3.89s
704:	learn: 0.0761995	total: 9.26s	remaining: 3.87s
705:	learn: 0.

858:	learn: 0.0596819	total: 11.3s	remaining: 1.85s
859:	learn: 0.0596200	total: 11.3s	remaining: 1.84s
860:	learn: 0.0594934	total: 11.3s	remaining: 1.83s
861:	learn: 0.0594344	total: 11.3s	remaining: 1.81s
862:	learn: 0.0593758	total: 11.3s	remaining: 1.8s
863:	learn: 0.0593269	total: 11.4s	remaining: 1.79s
864:	learn: 0.0592095	total: 11.4s	remaining: 1.77s
865:	learn: 0.0591059	total: 11.4s	remaining: 1.76s
866:	learn: 0.0590144	total: 11.4s	remaining: 1.75s
867:	learn: 0.0589751	total: 11.4s	remaining: 1.73s
868:	learn: 0.0589173	total: 11.4s	remaining: 1.72s
869:	learn: 0.0588570	total: 11.4s	remaining: 1.71s
870:	learn: 0.0587598	total: 11.4s	remaining: 1.7s
871:	learn: 0.0586638	total: 11.5s	remaining: 1.68s
872:	learn: 0.0585871	total: 11.5s	remaining: 1.67s
873:	learn: 0.0585032	total: 11.5s	remaining: 1.66s
874:	learn: 0.0583913	total: 11.5s	remaining: 1.64s
875:	learn: 0.0583076	total: 11.5s	remaining: 1.63s
876:	learn: 0.0582331	total: 11.5s	remaining: 1.62s
877:	learn: 0.

In [22]:
score

{'lr': [0.4171282893401066, 0.4148685152698367, 0.41844941980616324],
 'rf': [0.7143240189224906, 0.6760269848367688, 0.6868858494811341],
 'xgb': [0.7276484498428923, 0.6926221894389606, 0.7210574392657518],
 'bb': [0.6959097738478257, 0.6687138632244316, 0.67767346944216],
 'cc': [0.7143784443299563, 0.6701537006904437, 0.6887547987354881],
 'gn': [0.35739483181065745, 0.3326157539798046, 0.3487198495764282]}

In [32]:
# improve xgboost
score_dict = {}
dep_list = np.arange(16,26,2)
n_estimators_list = np.arange(160,260,20)
idx = 0
for data in [df1,df2,df3]:
    idx += 1
    for dep in dep_list:
        for es in n_estimators_list:
            X1= data.drop(['class'], axis=1)
            Y1 = data['class']
            X1_train, X1_test, Y1_train, Y1_test = train_test_split(X1, Y1, test_size = 0.2)
            xgb = XGBClassifier(max_depth=dep, n_estimators=es,objective='multi:softmax',num_class=3)
            xgb.fit(X1_train, Y1_train)
            y_predict = xgb.predict(X1_test)
            score_dict[len(score_dict)] = {'dep':dep, 'es':es, 'score':cohen_kappa_score(Y1_test,y_predict)}

In [36]:
score_df = pd.DataFrame(data = score_dict).T

In [45]:
score_df['data'] = ['df1','df2','df3']*25

In [57]:
for i in range(len(score_df)):
    if score_df['score'][i] == max(score_df['score']):
        print(score_df.iloc[i])

dep            24
es            240
score    0.737311
data          df1
Name: 24, dtype: object


Therefore, we need to set the XGBoost with maximum depth = 24 and number of estimators will 240.